src/core/datagram.c

```diff
@@ -31,6 +31,15 @@ QuicDatagramValidate(
     _In_ const QUIC_DATAGRAM* Datagram
     )
 {
+    QUIC_CONNECTION* Connection = QuicDatagramGetConnection(Datagram);
+    // If Datagram flag is set, SendQueue must not be null,
+    // Otherwise SendQueue must be null
```

Comment:
For one, our commenting format requires extra `//` at the beginning and end of the comment. For instance:
```
    //
    // If Datagram flag is set, SendQueue must not be null,
    // Otherwise SendQueue must be null
    //
```
Additionally, try to max out the 80 character limit before making a new line.

Finally, in this specific case, the comment would be better to say "why" and not "what" is going on. The code is self-explanatory for the "what" already. Your comment essentially is just repeating the code.